dashboard/app/models/user.rb

```diff
@@ -1569,7 +1569,7 @@ def assigned_script?(script)
 
   # Returns the set of courses the user has been assigned to or has progress in.
   def courses_as_student
-    scripts.map(&:unit_group).compact.concat(section_courses).uniq
+    visible_scripts.map(&:unit_group).compact.concat(section_courses).uniq
```

Comment:
Couldn't students have progress in one of the scripts where published_state is null? Based on the description of this function, I'd expect those to come back here

Reply:
Yes, but `get_published_state` returns the published state of the unit group if the script doesn't have one. See https://github.com/code-dot-org/code-dot-org/blob/73ac66a5a9998a75ba3c565da6ef2ea697c90a1d/dashboard/app/models/script.rb#L1803. So `get_published_state` for scripts like cds3-2021 should return `stable` as that's the published state of csd-2021.

Reply:
To expand on this a bit, we made the published state of scripts within unit group null for a couple reasons: to support CSA needing to only pilot some of the scripts within their unit group and because the field was already duplicated. The latter point meant that we had code that copied that published state from the unit group to each script on save, which we were able to get rid of when we made the script published_state null. `get_published_state` is the method that is supposed to handle all that logic, so wherever possible that method should be called instead of referencing `published_state` directly. This bug happened because I missed a spot when replacing `published_state` with `get_published_state`

Reply:
Makes sense! Thanks for that explanation